### ISSUE TITLE
Update default styles so that <custom-video> behavior mirrors <video>

### ISSUE THREAD

**Issue by dylanjha** [muxinc/custom-video-element#5]
When using `<custom-video>` -- or a subclass of `<custom-video>` for that matter -- the shadow dom is inserting `<styles>` that make the element deviate from the behavior of `<video>`.

For example, given code like this I would expect the elements to look visually identical:

```html
<style>
  video,
  custom-video {
     width: 100%;
     max-width: 500px;
     border: 1px solid lightcoral;
  }
</style>

<video
  controls
  muted
  src="https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe/high.mp4"
></video>

<custom-video
  src="http://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe/high.mp4"
  controls
  muted
>
</custom-video>
```

Here's [a code sandbox](https://i18gb.csb.app/) where you can see the height on `<custom-video>` is getting maxed out at 150px, and I actually can't find a good way around that without editing the styles of the `<video>` element inside the shadow dom (removing `height: 100%;`)

**Comment by heff** [comment]
The video element automatically resizes itself based on the aspect ratio of video. I [made this work in video.js](https://youtu.be/0AEFvF2bHtw?t=2963), but it's a lot of extra effort and I'm not sure it's really worth it yet. I assume most real use cases are fine defining the width and height, and not relying on the media el to figure that out. Do you have a use case where this is really valuable? There may be a way we can make this work with a default ratio at least using the padding approach.

**Comment by heff** [comment]
Ugh...I watched the video and young heff convinced old heff this would be a cool feature. At least some of it. We'll need it in media-container maybe more though.

**Comment by dylanjha** [comment]
I had a feeling this might be the case. Neat throwback video! Glad you have addressed this before and given it some thought.

I have some thoughts about the shift and how that affects [Cumulative Layout Shift](https://web.dev/cls/). I'm thinking of a solution for `media-container` where we can work around that if the user provides an aspect ratio & a `layout` strategy like `responsive` or `fill` (inspired by [next/image](https://nextjs.org/docs/api-reference/next/image#layout)) then we can layout the container to be the right size and not cause a shift.

We can set that conversation aside for now, as it's more related to `media-controller`

----

With regards to this issue in `custom-video-element`

>  I assume most real use cases are fine defining the width and height. Do you have a use case where this is really valuable?

Personally, I'm not typically hard-coding a width and a height. An approach I usually take is like the minimal example above:

- `width: 100%` (so that on mobile phones it takes up the whole viewport horizontally)
- `max-width: 500px` (500 is just an example, it could be 800px or 1000px)

This is a very simple example of making a decently-responsive `<video>` element -- but I wasn't able to achieve that with `custom-video` using pure CSS.

The above approach assumes videos have a landscape aspect ratio (set the width and let the height take care of itself). With a portrait aspect ratio you have to do the opposite: set the height and let the width take care of itself [in stream.new we handle that by detecting if the video is vertical or not](https://github.com/muxinc/stream.new/blob/main/components/video-player.tsx#L180-L187).

**Comment by dylanjha** [comment]
Update that @cjpillsbury was able to find a way to pull this off by setting

template:

```css
width: auto;
height: auto;
```

and video:

```css
max-width: 100%;
max-height: 100%;
min-width: 100%;
min-height: 100%;
```

Here's a [diff view](https://gist.github.com/dylanjha/782ed9932f7c06a1e5f9a6e465786cfb/revisions) of the changes

This allows the template to respect the "intrinsic dimensions" of the video and the custom element itself can be styled the same way as one would style `video`

@heff we copied over CustomVideoElement with the above change in `muxinc/elements` and are using it there -- but it would be great to get that change pushed upstream into this package.

Technically, it is a breaking change to the default behavior so that has to be considered.

The other thing we ran into is the need for a CustomAudioElement :) -- a bit out of scope for this package but something like CustomMediaElement as a super-class could make sense.
